librosa/core/__init__.py

```diff
@@ -12,4 +12,12 @@
 from .notation import *  # pylint: disable=wildcard-import
 
 
-__all__ = [_ for _ in dir() if not _.startswith("_")]
+__all__ = []
+__all__ += convert.__all__
+__all__ += audio.__all__
+__all__ += spectrum.__all__
+__all__ += pitch.__all__
+__all__ += constantq.__all__
+__all__ += harmonic.__all__
+__all__ += fft.__all__
+__all__ += notation.__all__
```

Comment:
I don't think this is necessary.

`__all__` will be automatically updated by the imports at the top.  What line 15 did was remove all "private" imported objects (those starting with a `_`).  Your proposed change here does not do this, and without that functionality, we may as well just use the automatic updates implied by import, right?